fastapi_sqla/__init__.py

```diff
@@ -147,24 +149,98 @@ class Page(Collection, Generic[T]):
     meta: Meta
 
 
-def _query_count(session: Session, query: DbQuery) -> int:
-    """Default function used to count items returned by a query.
+DbQuery = Union[LegacyQuery, Select]
+QueryCount = Callable[[SqlaSession, DbQuery], int]
+QueryCountDependency = Callable[..., QueryCount]
+PaginateSignature = Callable[[DbQuery], Page[T]]
 
-    Default Query.count is slower than a manually written query could be: It runs the
-    query in a subquery, and count the number of elements returned:
 
-    See https://gist.github.com/hest/8798884
-    """
-    return query.count()
+def query_count(session: SqlaSession, query: DbQuery) -> int:
+    @singledispatch
+    def _query_count(query: DbQuery):  # pragma no cover
+        "Dispatch on registered functions based on `query` type."
+        raise NotImplementedError(
+            f"no _query_count registered for type {type(query)!r}"
+        )
 
+    @_query_count.register
```

Comment:
```py
    @_query_count.register
```

looks like a display error to me (I know it's valid but that was a little confusing at first, the `@_` 😆

Reply:
removed